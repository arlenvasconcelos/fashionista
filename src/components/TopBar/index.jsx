import Topbar from './TopBar'

export default Topbar