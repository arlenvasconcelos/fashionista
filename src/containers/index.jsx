import App from './App/App';
import Products from './Products/Products';
import Cart from './Cart/Cart';
import SingleProduct from './SingleProduct/SingleProduct';

export {App, Products, Cart, SingleProduct}