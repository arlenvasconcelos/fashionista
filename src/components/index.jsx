import CartDetails from './CartDetails/CartDetails';
import CartFooter from './CartFooter/CartFooter';
import CartHeader from './CartHeader/CartHeader';
import Drawer from './Drawer/Drawer';
import Loading from './Loading/Loading';
import NavSearch from './NavSearch/NavSearch';
import NoImage from './NoImage/NoImage';
import Notification from './Notification/Notification';
import Product from './Product/Product';
import ProductDetails from './ProductDetails/ProductDetails';
import ProductImage from './ProductImage/ProductImage';
import ProductPrice from './ProductPrice/ProductPrice';
import TopBar from './TopBar/TopBar';

export {
  CartDetails, CartFooter, CartHeader, Drawer, Loading,
  NavSearch, NoImage, Notification, Product, ProductDetails, 
  ProductImage, ProductPrice, TopBar,
}