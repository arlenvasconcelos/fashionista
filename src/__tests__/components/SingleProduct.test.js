import React from "react";
import { mount, shallow, render } from "enzyme";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route } from "react-router-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import createStore from "redux-mock-store";

import { SingleProduct } from "../../containers";
import { ProductImage, ProductDetails } from "../../components";

Enzyme.configure({ adapter: new Adapter() });

const product = {
  actual_price: "R$ 199,90",
  code_color: "20002605_613",
  color: "TAPEÃ‡ARIA",
  color_slug: "tapecaria",
  discount_percentage: "",
  image:
    "https://viniciusvinna.netlify.app/assets/api-fashionista/20002605_615_catalog_1.jpg",
  installments: "3x R$ 66,63",
  name: "VESTIDO TRANSPASSE BOW",
  on_sale: false,
  regular_price: "R$ 199,90",
  sizes: [
    { available: false, size: "PP", sku: "5807_343_0_PP" },
    { available: true, size: "P", sku: "5807_343_0_P" },
    { available: true, size: "M", sku: "5807_343_0_M" },
    { available: true, size: "G", sku: "5807_343_0_G" },
    { available: false, size: "GG", sku: "5807_343_0_GG" },
  ],
};

describe("Single Product Test", () => {
  const mockStore = createStore();
  const INITIAL_STATE = {
    products: {
      allProducts: [],
      filteredProducts: [],
      openedFilter: false,
      errorStatus: false,
      errorMessage: "Error message",
    },
    cart: {
      items: [],
      open: false,
    },
  };

  const store = mockStore(INITIAL_STATE);
  const wrapper = mount(
    <Router>
      <Provider store={store}>
        <Route>
          <SingleProduct product={product} />
        </Route>
      </Provider>
    </Router>
  );

  it("should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should find productImage", () => {
    expect(wrapper.find(ProductImage)).toBeTruthy();
  });

  it("should find productDetails", () => {
    expect(wrapper.find(ProductDetails)).toBeTruthy();
  });
});
