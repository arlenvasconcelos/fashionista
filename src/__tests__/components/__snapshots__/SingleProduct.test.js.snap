// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Product Test should match the snapshot 1`] = `"<div class=\\"container\\"><main class=\\"single-product\\"><a class=\\"single-product__icon\\" href=\\"/\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><line x1=\\"19\\" y1=\\"12\\" x2=\\"5\\" y2=\\"12\\"></line><polyline points=\\"12 19 5 12 12 5\\"></polyline></svg></a><div class=\\"product-image\\"><img src=\\"https://viniciusvinna.netlify.app/assets/api-fashionista/20002605_615_catalog_1.jpg\\" alt=\\"\\"></div><div class=\\"product-details\\"><div class=\\"details__content\\"><h4 class=\\"details__name\\">VESTIDO TRANSPASSE BOW</h4><p class=\\"details__style\\">Ref:20002605_613</p><p class=\\"details__color\\">Cor: Tape√ßaria</p><div class=\\"details__sizes\\"><button disabled=\\"\\" class=\\"sizes__item\\">PP</button><button class=\\"sizes__item\\">P</button><button class=\\"sizes__item\\">M</button><button class=\\"sizes__item\\">G</button><button disabled=\\"\\" class=\\"sizes__item\\">GG</button></div><div class=\\"product__price\\">R$ 199,90</div><p class=\\"details__installments\\">3x R$ 66,63 sem juros</p><div class=\\"details__button\\"><button>COMPRAR</button></div></div></div></main></div>"`;
